csvy/validators.py

```diff
@@ -52,28 +58,32 @@ def validate_read(header: dict[str, Any]) -> dict[str, Any]:
         The validated header.
 
     """
-    validated_header = {}
+    validated_header: dict[str, Any] = {}
     for key, value in header.items():
+        value_ = value.model_dump() if isinstance(value, BaseModel) else value
         if key in VALIDATORS_REGISTRY:
+            if not isinstance(value_, dict):
```

Comment:
Thinking about this, I suppose any `Mapping` type would work, not just `dict`s. The same is true for the input argument.

That said, users probably aren't going to be using other types of `Mapping` here all that often.

Reply:
And would `TypeError` maybe be more appropriate than `ValueError`?

Reply:
Yes to the second - I'll update. About the first, anything that can be unpacked with `**` would work. I was not sure if `Mapping`s in general allow this, so I sticked to `dict` that I know will work and, as you say, it is what most people will use.

Reply:
Changed in validate_header [08e187c](https://github.com/ImperialCollegeLondon/pycsvy/pull/167/commits/08e187cdd78b41506cb20c44e4dc472b22618d05)